### PR TITLE
Ce support for specifying database name

### PR DESCRIPTION
**Pull request by sharadregoti**

### CODE REVIEW

File: gateway/config/config.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ type Crud map[string]*CrudStub // The key here is the alias for database type
 // CrudStub holds the config at the database level
 type CrudStub struct {
 	Type            string                    `json:"type,omitempty" yaml:"type"` // database type
+	Name            string                    `json:"name,omitempty" yaml:"name"` // name of the logical database or schema name according to the database type
```
Comment (YourTechBud) [inline gateway/config/config.go:72]: We should call this field `database` or `dbName`

File: gateway/modules/crud/crud.go [hunk 2]
```diff
@@ -81,14 +80,23 @@ func (m *Module) SetHooks(hooks *model.CrudHooks, metricHook model.MetricCrudHoo
 	m.metricHook = metricHook
 }
 
-func (m *Module) initBlock(dbType utils.DBType, enabled bool, connection string) (Crud, error) {
+func (m *Module) initBlock(dbType utils.DBType, enabled bool, connection, name string) (Crud, error) {
```
Comment (YourTechBud) [inline gateway/modules/crud/crud.go:83]: Don't call it name, call it `dbName`

File: gateway/modules/crud/crud.go [hunk 3]
```diff
@@ -81,14 +80,23 @@ func (m *Module) SetHooks(hooks *model.CrudHooks, metricHook model.MetricCrudHoo
...
 	case utils.Mongo:
-		return mgo.Init(enabled, connection)
+		return mgo.Init(enabled, connection, name)
 	case utils.EmbeddedDB:
 		return bolt.Init(enabled, connection)
 	case utils.MySQL, utils.Postgres, utils.SQLServer:
-		return sql.Init(dbType, enabled, m.removeProjectScope, connection)
+		c, err := sql.Init(dbType, enabled, connection, name)
+		if err == nil && enabled {
+			if err := c.CreateDatabaseIfNotExist(context.Background(), name); err != nil {
```
Comment (YourTechBud) [inline gateway/modules/crud/crud.go:92]: We shouldn't use context.Background here.

File: gateway/modules/crud/crud.go [hunk 4]
```diff
@@ -57,7 +56,7 @@ type Crud interface {
 	RawQuery(ctx context.Context, query string, args []interface{}) (int64, interface{}, error)
 	GetCollections(ctx context.Context, project string) ([]utils.DatabaseCollections, error)
 	DeleteCollection(ctx context.Context, project, col string) error
```
Comment (YourTechBud) [inline gateway/modules/crud/crud.go:58]: We can remove the project parameter from all methods since nobody is using it

Comment (YourTechBud) [reply gateway/modules/crud/crud.go:58]: We can also remove it from all the exposed method in `operations.go` and `internal.go`

File: gateway/config/config.go (hunk 1 above)
Comment (sharadregoti) [reply gateway/config/config.go:72]: i 'll rename it to dbName,
i had given a generic name as it can be db name or schema name.

**Comment by YourTechBud** [comment]
@sharadregoti everything tested?

**Comment by sharadregoti** [comment]
yes

**Comment by YourTechBud** [comment]
Fixes #497

**Pull request merged**